js/main.js

```diff
@@ -34,5 +45,123 @@ function handleSave(event) {
 
   $imgPreview.setAttribute('src', 'images/placeholder-image-square.jpg');
 
+  $entriesDisplayed.insertBefore(journalEntry(entry), $entriesDisplayed.firstElementChild);
+
   $newEntry.reset();
+
+  $noEntriesMessage.classList.add('hidden');
+  showPage('entries');
+}
+
+function handleNew(event) {
+  showPage('entry-form');
+}
+
+function handleDOMLoad(event) {
+  var previousDataJSON = localStorage.getItem('data-JSON');
+  if (previousDataJSON) {
+    data = JSON.parse(previousDataJSON);
```

Comment:
You should do these actions in `data.js`, that is where all localStorage code should be